File: JAVA/2696. Minimum String Length After Removing Substrings.java
class Solution {
    public int minLength(String s) {
        Stack<Character>a=new Stack<Character>();
        for(char c:s.toCharArray()){
            if(!a.empty()){
                if((a.peek()=='A' && c=='B')||(a.peek()=='C' && c=='D'))
                a.pop();
                else
                a.push(c);
            }
            else
                a.push(c);
        }
        return a.size();
    }
}

File: C++/2696. Minimum String Length After Removing Substrings.cpp
class Solution {
public:
    int minLength(string s) {
        vector<char>a;
        bool t=true;
        int i,n;
        for(char c:s){
            a.push_back(c);
        }
        while(t==true){
            n=a.size()-1;
            t=false;
            for(i=n;i>0;i--){
                if((a[i]=='B' && a[i-1]=='A')||(a[i]=='D' && a[i-1]=='C'))
                {
                    a.erase(a.begin()+i);
                    a.erase(a.begin()+i-1);
                    t=true;
                }
            }
            
        }
        return a.size();
    }
};